### PR TITLE
Retry when error is raft proposal dropped

### PR DESCRIPTION
**Pull request by birdstorm**
If the error message is `Raft Proposal Dropped`, it means that the store is in the middle of a transition. It should be able resolved by a retry.

Signed-off-by: birdstorm <samuelwyf@hotmail.com>

### CODE REVIEW

File: src/main/java/org/tikv/common/operation/KVErrorHandler.java [hunk 1]
```diff
@@ -229,10 +229,14 @@ public boolean handleResponseError(BackOffer backOffer, RespT resp) {
...
       }
 
-      logger.warn(String.format("Unknown error %s for region [%s]", error.toString(), ctxRegion));
+      logger.warn(String.format("Unknown error %s for region [%s]", error, ctxRegion));
       // For other errors, we only drop cache here.
       // Upper level may split this task.
       invalidateRegionStoreCache(ctxRegion);
+      // retry if raft proposal is dropped, it indicates the store is in the middle of transition
+      if (error.getMessage().contains("Raft Proposal Dropped")) {
+        return true;
```
Comment (marsishandsome) [inline src/main/java/org/tikv/common/operation/KVErrorHandler.java:238]: `invalidateRegionStoreCache` should be called in this case?

Comment (birdstorm) [reply src/main/java/org/tikv/common/operation/KVErrorHandler.java:238]: it has been called above.

Comment (marsishandsome) [reply src/main/java/org/tikv/common/operation/KVErrorHandler.java:238]: i mean `invalidateRegionStoreCache` should not be called in this case.

Comment (birdstorm) [reply src/main/java/org/tikv/common/operation/KVErrorHandler.java:238]: it should be called, because the region info is incorrect.

**Review by marsishandsome — APPROVED** [review]
LGTM

**Pull request merged**